File: api/schemas/goals.py
from pydantic import BaseModel, Field
from typing import Optional


class GoalCreate(BaseModel):
    title: str
    description: Optional[str] = ""
    frequency_per_week: int = Field(ge=1, le=7)


class GoalUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    frequency_per_week: Optional[int] = Field(default=None, ge=1, le=7)
    frequency: Optional[int] = Field(default=None, ge=1, le=7)


class GoalIdResponse(BaseModel):
    id: int

File: src/shared/schemas/goals.ts
import { z } from 'zod';

export const goalCreateSchema = z.object({
  title: z.string().min(1),
  description: z.string().optional().default(''),
  frequency_per_week: z.number().int().min(1).max(7),
});

export const goalUpdateSchema = z.object({
  title: z.string().min(1).optional(),
  description: z.string().optional(),
  frequency_per_week: z.number().int().min(1).max(7).optional(),
  frequency: z.number().int().min(1).max(7).optional(),
});

export type GoalCreateInput = z.infer<typeof goalCreateSchema>;
export type GoalUpdateInput = z.infer<typeof goalUpdateSchema>;
